hilti/src/rt/tests/network.cc

```diff
@@ -0,0 +1,294 @@
+// Copyright (c) 2020 by the Zeek Project. See LICENSE for details.
+//
+#include "rt/types/address.h"
+#include <doctest/doctest.h>
+
+#include <hilti/rt/types/network.h>
+
+using namespace hilti::rt;
+
+TEST_SUITE_BEGIN("Network");
+
+TEST_CASE("comparison") {
+    const Address addr1("255.255.255.255");
+    const Address addr2("0.0.0.0");
+    const Address addr3("ffff:ffff:ffff:ffff:ffff:ffff:ffff:ffff");
+    const Address addr4("0:0:0:0:0:0:0:0");
+
+    CHECK_EQ(Network(addr1, 0), Network(addr1, 0));
+    CHECK_EQ(Network(addr1, 12), Network(addr1, 12));
+    CHECK_EQ(Network(addr1, 32), Network(addr1, 32));
+
+    CHECK_NE(Network(addr1, 32), Network(addr1, 0));
+    CHECK_NE(Network(addr1, 32), Network(addr2, 32));
+    CHECK_NE(Network(addr1, 0), Network(addr3, 0));
+    // TODO(bbannier): `Address` cannot distinguish the different families of these addresses.
```

Comment:
This and the other TODOs in this file need work on `rt::Address` to explicitly store its address family.

Reply:
Yeah, seems we need to store the family with address. I had been hoping to avoid that, but the results feel non-intuitive.